llm/src/speed_monitor_w_mfu.py

```diff
@@ -74,28 +69,57 @@ def get_gpu_flops_available(state):
     return gpu_flops_available
 
 
-class SpeedMonitorMFU(SpeedMonitor):
+class SpeedMonitorMFU(Callback):
     """Logs the training throughput and MFU.
     """
     def __init__(
         self,
         window_size: int = 100,
         gpu_flops_available: Union[float, int] = None
     ):
-        super().__init__(window_size=window_size)
-        self.set_gpu_flops_available = False
+        # Track the batch num samples and wct to compute throughput over a window of batches
+        self.batch_start_num_samples = 0
+        self.batch_start_wct = 0.0
+        self.batch_wct_buffer: Deque[float] = deque(maxlen=window_size)
+        self.batch_num_samples_buffer: Deque[int] = deque(maxlen=window_size)
+        self.window_size = window_size
         self.gpu_flops_available = gpu_flops_available
```

Comment:
I know you're fixing my mistakes but should't this just be 
```
if gpu_flops_available:
    self.gpu_flops_available = gpu_flops_available
else:
    self.gpu_flops_available = get_gpu_flops_available(state)
```
instead of setting `self.gpu_flops_available` in `before_dataloader()`

Reply:
The trouble is `state` is not available on `__init__` :(